### ISSUE TITLE
rebind.so not found

### ISSUE THREAD

**Issue by benjaminbours** [novnc/websockify#413]
Hello, sorry but I checked each issue about rebind.so but I cannot find a solution.

I'm on macOs and I used the Makefile and websockify is correctly installed in my path. But I always get this error when I use this command, like described in the exemple 
`websockify 2023 --wrap-mode=respawn -- telnetd -debug 2023`

`
Traceback (most recent call last):
  File "/usr/local/bin/websockify", line 9, in <module>
    load_entry_point('websockify==0.9.0', 'console_scripts', 'websockify')()
  File "/Library/Python/2.7/site-packages/websockify-0.9.0-py2.7.egg/websockify/websocketproxy.py", line 725, in websockify_init
    server = WebSocketProxy(**opts.__dict__)
  File "/Library/Python/2.7/site-packages/websockify-0.9.0-py2.7.egg/websockify/websocketproxy.py", line 306, in __init__
    raise Exception("rebind.so not found, perhaps you need to run make")
Exception: rebind.so not found, perhaps you need to run make
`

When I use the run script, I got kind of the same error but slightly different
`./run 2023 --wrap-mode=respawn -- telnetd -debug 2023`

`
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/Users/benjaminbours/Downloads/websockify-0.9.0/websockify/__main__.py", line 4, in <module>
    websockify.websocketproxy.websockify_init()
  File "websockify/websocketproxy.py", line 725, in websockify_init
    server = WebSocketProxy(**opts.__dict__)
  File "websockify/websocketproxy.py", line 306, in __init__
    raise Exception("rebind.so not found, perhaps you need to run make")
Exception: rebind.so not found, perhaps you need to run make
`

Is it related to this issue ? #156

**Comment by samhed** [comment]
What version of macOS are you trying to run websockify on?

**Comment by samhed** [comment]
No response

**Comment by Jousboxx** [comment]
I'm having what appears to be the same issue. I'm working on a system to automatically create Singularity containers for NoVNC, so my websockify command is a bit convoluted:
` websockify -D --web=/usr/share/novnc/ --cert=/etc/ssl/novnc.pem $SINGULARITY_NOVNC_PORT localhost:$SINGULARITY_VNC_PORT -- vncserver -rfbport $SINGULARITY_VNC_PORT`

Here's the error message:
``` Traceback (most recent call last):
  File "/usr/bin/websockify", line 11, in <module>
    load_entry_point('websockify==0.9.0', 'console_scripts', 'websockify')()
  File "/usr/lib/python3/dist-packages/websockify/websocketproxy.py", line 725, in websockify_init
    server = WebSocketProxy(**opts.__dict__)
  File "/usr/lib/python3/dist-packages/websockify/websocketproxy.py", line 306, in __init__
    raise Exception("rebind.so not found, perhaps you need to run make")
Exception: rebind.so not found, perhaps you need to run make 
```

I've seen a `run` script mentioned in a lot of places, but can't find it so haven't tried @benjaminbours 's second method.

OS: Ubuntu 20.04 (Host and container)
Websockify version: 0.9.0

**Comment by snickell** [comment]
@samhed fyi I'm seeing the same issue as @aerobinsonIV above. Also with ubuntu 20.04, probably a new issue, but his report is so well put together, I wonder if its worth re-opening the issue? The general theme is that the mechanism for finding rebind.so is somewhat fragile.

**Comment by aerobinsonIV** [comment]
@snickell Are you using a custom or old build of noVNC? That's what I was doing when I first encountered the issue. I fixed the problem temporarily by downloading noVNC 1.2.0 from https://github.com/novnc/noVNC/releases and manually copying everything into /usr/share/novnc.

The current (working) version of my project uses noVNC 1.0.0 from the Ubuntu 20.04 repositories, which is strange given that that's probably the version you're on. This suggests that it's not a bug with noVNC.

Unfortunately the problem resolved itself while I was messing with other stuff, so I don't know exactly what the long-term solution was. However, I'd be happy to share details about my working setup. Feel free to email me, aerobinsoniv@gmail.com
